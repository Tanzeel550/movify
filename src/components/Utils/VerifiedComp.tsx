import React, { ReactElement } from 'react';
import VerifyEmailBtn from './VerifyEmailBtn';

type Props = {
  children:
    | React.ReactNode
    | ReactElement<any, any>
    | JSX.Element
    | JSX.Element[];
  // children: Component<{}, {}, any>;
  emailVerified: boolean;
};

const VerifiedComp: React.FC<Props> = ({ children, emailVerified }: Props) => {
  return emailVerified ? (
    <React.Fragment>{children}</React.Fragment>
  ) : (
    <div className="container jumbotron">
      <p>email is ${emailVerified}</p>
      <h3 className="u-text-colorized">
        You cannot access this page because your Email is not verified
      </h3>
      <VerifyEmailBtn />
    </div>
  );
};

export default VerifiedComp;
