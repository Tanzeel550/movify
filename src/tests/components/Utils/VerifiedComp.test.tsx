import { shallow } from 'enzyme';
import VerifiedComp from '../../../components/Utils/VerifiedComp';

import React from 'react';

const Comp = () => <h1>Hello from Comp </h1>;

test('should render the component because the email is verified', () => {
  const wrapper = shallow(
    <VerifiedComp emailVerified={true} children={<Comp />} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should not render the component because the email is not verified', () => {
  const wrapper = shallow(
    <VerifiedComp emailVerified={false} children={<Comp />} />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('h3').text()).toMatch(
    'You cannot access this page because your Email is not verified'
  );
});

export {};
