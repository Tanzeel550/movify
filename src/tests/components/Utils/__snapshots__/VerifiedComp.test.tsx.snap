// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render the component because the email is not verified 1`] = `
<div
  className="container jumbotron"
>
  <p>
    email is $
  </p>
  <h3
    className="u-text-colorized"
  >
    You cannot access this page because your Email is not verified
  </h3>
  <Connect(VerifyEmailBtn) />
</div>
`;

exports[`should render the component because the email is verified 1`] = `
<Fragment>
  <Comp />
</Fragment>
`;
